src/Core/Checkout/Cart/Event/CartChangedEvent.php

```diff
@@ -3,38 +3,33 @@
 namespace Shopware\Core\Checkout\Cart\Event;
 
 use Shopware\Core\Checkout\Cart\Cart;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\Event\ShopwareSalesChannelEvent;
 use Shopware\Core\Framework\Log\Package;
 use Shopware\Core\System\SalesChannel\SalesChannelContext;
 use Symfony\Contracts\EventDispatcher\Event;
 
 #[Package('checkout')]
-class CartChangedEvent extends Event
+class CartChangedEvent extends Event implements ShopwareSalesChannelEvent, CartEvent
 {
-    /**
-     * @var Cart
-     */
-    protected $cart;
-
-    /**
-     * @var SalesChannelContext
-     */
-    protected $context;
-
     public function __construct(
-        Cart $cart,
-        SalesChannelContext $context
+        protected readonly Cart $cart,
+        protected readonly SalesChannelContext $salesChannelContext,
```

Comment:
This is a breaking change. Any code that uses named parameters will break now. You also changed the properties from `protected` to `protected readonly` which means if any plugin extends this class, and tries to change those properties, it will break.

Reply:
I don't believe this is something someone would do. It's also forbidden on the other elements.
So while I can agree this is a breaking change, I'm quite confident it won't break anything.

Shall I leave out the `readonly` on this event class?

Reply:
I'm not sure what you mean with forbidden in other events. If events were introduced with `readonly` properties, or followed the correct deprecation flow, then them being forbidden has no bearing on what happens here.

I'm fine with removing `readonly`, but `$salesChannelContext` should remain `$context`. You can add a deprecation stating that it will be renamed.

```
    /**
     * @deprecated tag:v6.7.0 - Param $context will be renamed to $salesChannelContext
     */
```

Reply:
I reverted the readonly addition and the attribute name change.

Reply:
And now I've also added this attribute deprecation.

Reply:
BC is still complaining
```
[BC] CHANGED: Type of property Shopware\Core\Checkout\Cart\Event\CartChangedEvent#$cart changed from having no type to Shopware\Core\Checkout\Cart\Cart
[BC] CHANGED: Type of property Shopware\Core\Checkout\Cart\Event\CartChangedEvent#$context changed from having no type to Shopware\Core\System\SalesChannel\SalesChannelContext
2 backwards-incompatible changes detected
```

Is this acceptable or shall I also revert that?

Reply:
Reverted that too. I hope I left obvious comments what's to be done in 6.7 when removing deprecations!